tekton/utils/pipeline.go

```diff
@@ -26,24 +35,35 @@ type PipelineRef struct {
 
 	// Params is a slice of parameters for a given resolver
 	Params []Param `json:"params"`
+}
+
+// Pipeline contains a reference to a Pipeline and the name of the service account to use while executing it.
+// +kubebuilder:object:generate=true
+type Pipeline struct {
+	// PipelineRef is the reference to the Pipeline
+	PipelineRef PipelineRef `json:"pipelineRef"`
+
+	// ServiceAccount is the ServiceAccount to use during the execution of the Pipeline
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +optional
+	ServiceAccount string `json:"serviceAccountName,omitempty"`
 
 	// Timeout is value to use to override the tekton default Pipelinerun timeout
 	// +kubebuilder:default="0"
 	// +optional
 	Timeout string `json:"timeout,omitempty"`
 }
 
-// Param defines the parameters for a given resolver in PipelineRef
-type Param struct {
-	// Name is the name of the parameter
-	Name string `json:"name"`
-
-	// Value is the value of the parameter
-	Value string `json:"value"`
+// ParametrizedPipeline is an extension of the Pipeline struct, adding an array of parameters that will be passed to
+// the Pipeline.
+// +kubebuilder:object:generate=true
+type ParametrizedPipeline struct {
```

Comment:
Can we call it ParameterizedPipeline? Parametrized isn't a word

Reply:
Fixed.